### PR TITLE
Use PJ in debug signal handling

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Seems bare JSON printing left in some places, see https://github.com/rizinorg/rizin/issues/200#issuecomment-748905974
Converting the debug signals handling to PJ API.
Also added a simple paragraph on how to use PJ API in developers' documentation.

**Test plan**

CI is green. See also `dkj` command output.

**Closing issues**

Partially addresses https://github.com/rizinorg/rizin/issues/200

...

### CODE REVIEW

File: DEVELOPERS.md [hunk 1]
```diff
@@ -249,3 +249,34 @@ Hint: To find both the declaration and definition of a function named
 ```bash
 git grep -nWG "^[^[:blank:]].*func_name("
 ```
+
+## JSON
```
Comment (ret2libc) [inline DEVELOPERS.md:253]: Awesome idea! I would add also the final JSON printed by the following code, just to be sure.

File: librz/debug/dsignal.c [hunk 2]
```diff
@@ -85,43 +85,58 @@ static bool siglistcb(void *p, const char *k, const char *v) {
...
-		if (dbg->_mode == 2) {
-			dbg->_mode = 0;
-		} else {
-			rz_cons_strcat(",");
+		opt = (int)sdb_num_get(dpj->dbg->sgnls, key, 0);
+		if (dpj->dbg->_mode == 2) {
+			dpj->dbg->_mode = 0;
 		}
-		rz_cons_printf("{\"signum\":\"%s\",\"name\":\"%s\",\"option\":", k, v);
+		//} else {
```
Comment (ret2libc) [inline librz/debug/dsignal.c]: remove

File: librz/debug/dsignal.c [hunk 3]
```diff
@@ -85,43 +85,58 @@ static bool siglistcb(void *p, const char *k, const char *v) {
 	return true;
 }
 
+struct DebugPJ {
```
Comment (ret2libc) [inline librz/debug/dsignal.c]: I would use `debug_pj`, as in general we use those names unless it is for exported things.

File: DEVELOPERS.md (hunk 1 above)
Comment (XVilka) [reply DEVELOPERS.md:253]: Added.

**Review by ret2libc — APPROVED** [review]

**Pull request merged**